compile.sh

```diff
@@ -35,6 +35,51 @@ if [[ $EUID != 0 && "$1" != vagrant ]]; then
 	exit $?
 fi
 
+update_src() {
+	cd "${SRC}" || exit
+	if [[ ! -f $SRC/.ignore_changes ]]; then
+		echo -e "[\e[0;32m o.k. \x1B[0m] This script will try to update"
+		git pull
+		CHANGED_FILES=$(git diff --name-only)
+		if [[ -n $CHANGED_FILES ]]; then
+			echo -e "[\e[0;35m warn \x1B[0m] Can't update since you made changes to: \e[0;32m\n${CHANGED_FILES}\x1B[0m"
+			while true; do
+				echo -e "Press \e[0;33m<Ctrl-C>\x1B[0m or \e[0;33mexit\x1B[0m to abort compilation, \e[0;33m<Enter>\x1B[0m to ignore and continue, \e[0;33mdiff\x1B[0m to display changes"
+				read -r
+				if [[ "$REPLY" == "diff" ]]; then
+					git diff
+				elif [[ "$REPLY" == "exit" ]]; then
+					exit 1
+				elif [[ "$REPLY" == "" ]]; then
+					break
+				else
+					echo "Unknown command!"
+				fi
+			done
+		else
+			git checkout "${LIB_TAG:-master}"
+		fi
+	fi
+}
+
+TMPFILE=`mktemp`
+echo SRC=$SRC > $TMPFILE
+echo LIB_TAG=$LIB_TAG >> $TMPFILE
+declare -f update_src >> $TMPFILE
+echo update_src >> $TMPFILE
+
+#do not update/checkout git with root privileges to messup files onwership.
```

Comment:
But only inside Docker. For native it shall remain unchanged. What do you think?

Reply:
no.  for bare metal, it should not use root to git pull/checkout, otherwise wrong file ownership.

Reply:
If we don't pull, we don't know if there are changes upstream. Sadly we need root / privileged mode for building images. Not for kernel.

Related?
https://forum.armbian.com/topic/11747-q-bash-cannot-set-terminal-process-group-at-compilesh/